### PR TITLE
Use RBE when prow runs with workload identity.

### PR DESCRIPTION
**Pull request by bianpengyuan**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: prow/proxy-common.inc [hunk 1]
```diff
@@ -68,4 +72,17 @@ if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
       export BAZEL_BUILD_ARGS="${BAZEL_BUILD_ARGS} --google_credentials=${GOOGLE_APPLICATION_CREDENTIALS} --remote_instance_name=${BAZEL_BUILD_RBE_INSTANCE} --remote_cache=${BAZEL_BUILD_RBE_CACHE}"
     fi
   fi
+elif curl -H "${METADATA_QUERY_HEADER}" "${METADATA_SERVER_URL}" | grep "${WORKLOAD_IDENTITY_SUFFIX}"; then
```
Comment (howardjohn) [inline prow/proxy-common.inc:75]: i think it's bound to a GSA so may need to change the check to account for that? `gcloud auth list --filter=status:ACTIVE --format="value(account)"` may be simpler as well

File: prow/proxy-common.inc [hunk 2]
```diff
@@ -50,14 +50,18 @@ export BAZEL_OUT
 # Disable RBE execution until we have LARGE_MACHINE instances provisioned for building V8.
 export BAZEL_BUILD_RBE_JOBS="${BAZEL_BUILD_RBE_JOBS:-0}"
```
Comment (howardjohn) [inline prow/proxy-common.inc:51]: nit- the comment here is outdated, the new issue is the tool chain used in RBE is too new for us. We will need to see if we can switch it over to our image

**Comment by bianpengyuan** [comment]
Let me dig a bit into the failure..

**Review by howardjohn — APPROVED** [review]

**Review by kyessenov — APPROVED** [review]

**Pull request merged**